firmware/wpilib-ws-romi/wpilib-ws-romi.ino

```diff
@@ -42,29 +42,89 @@ uint8_t ioChannelModes[5] = {kModeDigitalOut, kModeDigitalOut, kModeDigitalOut,
 uint8_t ioDioPins[5] = {11, 4, 20, 21, 22};
 uint8_t ioAinPins[5] = {0, A6, A2, A3, A4};
 
-bool dio8IsInput = false;
-
+bool isTestMode = false;
 bool isConfigured = false;
 
 void setup() {
   rPiLink.init(20);
 
-  pwms[0].attach(21);
-  pwms[1].attach(22);
+  // Determine if we should enter test mode
+  // If button A and B are pressed during power up, enter test mode
+  // TODO Implement
+  if (buttonA.isPressed() && buttonB.isPressed()) {
+    isTestMode = true;
+  }
+
+  if (isTestMode) {
+    testModeInit();
+  }
+  else {
+    normalModeInit();
+  }
+}
 
+// Initialization routines for test mode
+void testModeInit() {
+  buzzer.play("!L16 v10 cdefgab>c");
+
+  Serial.begin(9600);
+  while (!Serial) {}
+
+}
+
+// Initialization routines for normal operation
+void normalModeInit() {
   buzzer.play("v10>>g16>>>c16");
 }
 
 unsigned long lastHeartbeat = 0;
 
-void loop() {
-  // Get the latest data including recent i2c master writes
-  rPiLink.updateBuffer();
+void testModeLoop() {
+  // Used to verify mode settings
+  uint16_t ioConfig = rPiLink.buffer.ioConfig;
```

Comment:
I don't think we need a pointer (adds complexity) since it's all simple primitive types. We set it back to 0 so that we don't constantly run the configuration routine (since we check for the modified flag in the register, which is the MSB)